### ISSUE TITLE
Figure out why steps have different results when a checkpoint is used

### ISSUE THREAD

**Issue by mmaecki** [SebChw/Actually-Robust-Training#137]
Figure out why steps have different results when a checkpoint is used - i.e. Overfit train set does not pass the check at the first iteration of tutorial, however it does, when any checkpoint is being used before this step.

**Comment by SebChw** [comment]
This is just because Random generators are in different places as we skip some other steps

**Comment by SebChw** [comment]
This has influence on:
* Data shuffling process in train dataloader
* Initial weights of the network

I couldn't find any better idea nor a bug in the code
